src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -55,9 +55,20 @@ private[avro] class AvroRelation(
   private val recordNamespace = parameters.getOrElse("recordNamespace", "")
 
   /** needs to be lazy so it is not evaluated when saving since no schema exists at that location */
-  private lazy val avroSchema = paths match {
-    case Array(head, _*) => newReader(head)(_.getSchema)
-    case Array() => throw NoFilesException
+  private lazy val avroSchema = if (paths.isEmpty) {
+    throw NoFilesException
+  } else {
+    // As of Spark 1.5.0, it's possible to receive an array which contains a single non-existent
```

Comment:
Yeah, according existing code paths calling `globPathIfNecessary` in Spark 1.5, this assumption is right. We should probably do the existence check within `globPathIfNecessary` when the path pattern doesn't contain any glob characters though.